Validation/Geometry/src/MaterialBudgetHcal.cc

```diff
@@ -36,10 +38,17 @@ MaterialBudgetHcal::MaterialBudgetHcal(const edm::ParameterSet& p) {
 void MaterialBudgetHcal::update(const BeginOfJob* job) {
   //----- Check that selected volumes are indeed part of the geometry
   // Numbering From DDD
-  edm::ESTransientHandle<DDCompactView> pDD;
-  (*job)()->get<IdealGeometryRecord>().get(pDD);
-  if (theHistoHcal_)
-    theHistoHcal_->fillBeginJob((*pDD));
+  if (fromdd4hep_) {
+    edm::ESTransientHandle<cms::DDCompactView> pDD;
+    (*job)()->get<IdealGeometryRecord>().get(pDD);
+    if (theHistoHcal_)
+      theHistoHcal_->fillBeginJob((*pDD));
+  } else {
+    edm::ESTransientHandle<DDCompactView> pDD;
+    (*job)()->get<IdealGeometryRecord>().get(pDD);
+    if (theHistoHcal_)
+      theHistoHcal_->fillBeginJob((*pDD));
+  }
```

Comment:
Both if/else options do the same?

Reply:
Yes they do - but the 2 pDD's are different type of objects DDCompactView vs cms::DDCompactView - one works for DDD and the other for DD4Hep. I think this may be the simplest construct.

Reply:
OK, I see. Thanks